### PR TITLE
Add Auth0. Resolves #26

### LINKED ISSUE #26: Implement user authentification with Auth0
The #3 task should be completed to start this task.
Create and use pages Login and Logout. 
Request Auth0 login endpoint, after successful authentification receive data and store this state and user data.
Use UserProfile page and render authorized user data (for now let's deal with data that we only receive from Auth0).
Reflect logged-in status in the SiteTopNavigation.

Similar task in backend is https://github.com/pets-oss/pets-back/issues/22. You might need some coordination and help.

### PR DESCRIPTION
**Pull request by benNek**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,2 +0,0 @@
-REACT_APP_GRAPHQL_URL=https://petbook-back.herokuapp.com/graphql
```
Comment (benNek) [inline .env:1]: .env files shouldnt be commited. soo when pulling don't forget to add `REACT_APP_AUTH0_DOMAIN` and `REACT_APP_AUTH0_CLIENT_ID` with the respective values from auth0 configuration dashboard

**Comment by benNek** [comment]
Fixing tests 😬

**Review by rikius — APPROVED** [review]

**Comment by rikius** [comment]
Also, it will be nice to cover all with the test this implementation.

File: .env (hunk 1 above)
Comment (domzic) [reply .env:1]: For local overrides we should use .env.local. (which is in gitignore).
IMO `.env` can contain non secret values. For secrets better to put empty value, like either
`REACT_APP_AUTH0_CLIENT_ID=`
or
`REACT_APP_AUTH0_CLIENT_ID=***secret***`

So devs could see what secrets they need to override in local env file and you won't need to tell explicitly after each commit

File: .gitignore [hunk 2]
```diff
@@ -24,3 +24,6 @@ yarn-error.log*
 
 .eslintcache
 .idea
+.env
+.env.local
+.env.staging
```
Comment (domzic) [inline .gitignore:29]: `.env.staging` ?

File: src/components/layout/SiteTopNavigation.tsx [hunk 3]
```diff
@@ -79,11 +80,38 @@ function ListItemLink(props: ListItemLinkProps) {
...
+            <ListItem className={className} button onClick={onClick}>
+                {icon ? <ListItemIcon>{icon}</ListItemIcon> : null}
+                {primary ? <ListItemText primary={primary} /> : null}
+            </ListItem>
+        </li>
+    );
+}
+
+function AuthNav() {
+    return (
```
Comment (domzic) [inline src/components/layout/SiteTopNavigation.tsx:104]: `return <ListItemLink to="/animal-list" primary="Animals" />`

File: src/utils/auth/Auth0ProviderWithHistory.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import React, { ReactChild } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { Auth0Provider } from '@auth0/auth0-react';
+
+const Auth0ProviderWithHistory = ({ children }: { children: ReactChild }) => {
+    const domain = process.env.REACT_APP_AUTH0_DOMAIN as string;
```
Comment (domzic) [inline src/utils/auth/Auth0ProviderWithHistory.tsx:7]: Put configs to `src/config.ts`

**Review by domzic — APPROVED** [review]

**Review by domzic — APPROVED** [review]

**Pull request merged**